pkg/test/masteruserrecord/assertion.go

```diff
@@ -73,6 +73,15 @@ func (a *Assertion) AllUserAccountsHaveStatusSyncIndex(syncIndex string) *Assert
 	return a
 }
 
+func (a *Assertion) AllUserAccountsHaveCluster(expected toolchainv1alpha1.Cluster) *Assertion {
+	err := a.loadUaAssertion()
+	require.NoError(a.t, err)
+	for _, ua := range a.masterUserRecord.Status.UserAccounts {
+		assert.Equal(a.t, expected, ua.Cluster)
```

Comment:
why don't you use something like `ua.Cluster.Name == targetCluster || ua.TargetCluster == targetCluster` as above?

Reply:
Because we need to compare the entire cluster (which is a stuct now, not just a string). Comparing cluster names only is not enough. And we don't care about deprecated `TargetCluster` here because this new assertion is used in the new code only.

Reply:
ok, so the deprecated `TargetCluster` field will be empty for both `expected` and `ua` objects in this case, then. Makes sense.